### PR TITLE
feat(presentational): update button component to use hoc

### PR DESCRIPTION
**Pull request by agory**

### CODE REVIEW

**Comment by mfrachet** [comment]
Tests are not passing

**Comment by agory** [comment]
Sorry, I made a little mistake. I took the opportunity to add hoc size :smile: .

File: src/web/components/Button/Button.test.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
 import React from 'react';
 import { shallow } from 'enzyme';
-import Button from './';
+import { ButtonBase } from './';
 
 describe('Button', () => {
```
Comment (mfrachet) [inline src/web/components/Button/Button.test.js]: ButtonBase I guess so ?

File: src/web/components/Button/index.js [hunk 2]
```diff
@@ -1,25 +1,16 @@
...
-  onClick,
-  color,
-  disabled,
-  size,
-  outlined,
-  inverted,
-  rounded,
-  loading,
-  className,
+function Button({
```
Comment (mfrachet) [inline src/web/components/Button/index.js]: ButtonBase too ?

File: src/web/components/Button/index.js [hunk 3]
```diff
@@ -46,12 +35,13 @@ Button.propTypes = {
...
   className: '',
-  size: '',
   disabled: false,
   outlined: false,
   inverted: false,
   rounded: false,
   loading: false,
 };
+
+const ColorButton = sizable(colorify(Button));
```
Comment (mfrachet) [inline src/web/components/Button/index.js]: `ColoredButton`

File: src/web/components/Button/Button.test.js [hunk 4]
```diff
@@ -1,14 +1,14 @@
 import React from 'react';
 import { shallow } from 'enzyme';
-import Button from './';
+import { ButtonBase } from './';
```
Comment (mfrachet) [inline src/web/components/Button/Button.test.js]: The fileName is always `Button.test` and not `ButtonBase.spec` 😛

Comment (agory) [reply src/web/components/Button/Button.test.js]: Sorry :pray:

**Comment by mfrachet** [comment]
storybook doesn't start on this PR

**Comment by agory** [comment]
With a rebase was solved. The problem was linked to an hoc that has bien updated in another pr.

**Comment by mfrachet** [comment]
Always missing storybook on this one 😝

**Comment by mfrachet** [comment]
Icons before and after seems weird on the storybook 🤔

**Pull request closed without merging**